base/exporter.go

```diff
@@ -0,0 +1,318 @@
+		}
+		break
+	default:
+		panic("Err")
+	}
+
+}
+
+func (s *SgwIntStat) Set(newV interface{}) {
+	s.Value = newV
+	fmt.Println(s.Value)
+}
+
+func (s *SgwIntStat) SetIfMax(newV int) {
+	switch formattedVal := s.Value.(type) {
+	case int:
+		if newV > formattedVal {
+			s.Value = newV
+		}
+		break
+	default:
+		panic("err")
+	}
+}
+
+func (s *SgwIntStat) Add(newV int) {
+
+	if s.Value == nil {
+		s.Value = newV
+	}
+
+	switch formattedVal := s.Value.(type) {
+	case int:
+		s.Value = formattedVal + newV
+		break
+	default:
+		panic("Err")
+	}
+}
+
+func (s *SgwIntStat) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf("%v", s.Value)), nil
+}
+
+func (s *SgwIntStat) String() string {
+	return fmt.Sprintf("%v", s.Value)
+}
+
+type QueryStat struct {
+	QueryCount      *SgwIntStat
+	QueryErrorCount *SgwIntStat
+	QueryTime       *SgwIntStat
+}
+
+func (s *SgwStats) DBStats(name string) *DbStats {
+
+	if s.DbStats == nil {
+		s.DbStats = map[string]*DbStats{}
+	}
+
+	if _, ok := s.DbStats[name]; !ok {
+		s.DbStats[name] = &DbStats{}
+	}
+	return s.DbStats[name]
+}
+
+func (d *DbStats) DeltaSync(dbName string) *DeltaSyncStats {
+	if d.DeltaSyncStats == nil {
+		d.DeltaSyncStats = &DeltaSyncStats{
+			DeltasRequested:           NewIntStat("delta_sync", "deltas_requested", dbName, prometheus.CounterValue, 0),
+			DeltasSent:                NewIntStat("delta_sync", "deltas_sent", dbName, prometheus.CounterValue, 0),
+			DeltaPullReplicationCount: NewIntStat("delta_sync", "delta_pull_replication_count", dbName, prometheus.CounterValue, 0),
+			DeltaCacheHit:             NewIntStat("delta_sync", "delta_cache_hit", dbName, prometheus.CounterValue, 0),
+			DeltaCacheMiss:            NewIntStat("delta_sync", "delta_sync_miss", dbName, prometheus.CounterValue, 0),
+			DeltaPushDocCount:         NewIntStat("delta_sync", "delta_push_doc_count", dbName, prometheus.CounterValue, 0),
+		}
+	}
+	return d.DeltaSyncStats
+}
+
+func (d *DbStats) SharedBucketImport() *SharedBucketImportStats {
+	if d.SharedBucketImportStats == nil {
+		dbName := d.dbName
+		d.SharedBucketImportStats = &SharedBucketImportStats{
+			ImportCount:          NewIntStat("shared_bucket_import", "import_count", dbName, prometheus.CounterValue, 0),
+			ImportCancelCAS:      NewIntStat("shared_bucket_import", "import_cancel_cas", dbName, prometheus.CounterValue, 0),
+			ImportErrorCount:     NewIntStat("shared_bucket_import", "import_error_count", dbName, prometheus.CounterValue, 0),
+			ImportProcessingTime: NewIntStat("shared_bucket_import", "import_processing_time", dbName, prometheus.GaugeValue, 0),
+			ImportHighSeq:        NewIntStat("shared_bucket_import", "import_high_seq", dbName, prometheus.CounterValue, 0),
+			ImportPartitions:     NewIntStat("shared_bucket_import", "import_partitions", dbName, prometheus.GaugeValue, 0),
+		}
+	}
+
+	return d.SharedBucketImportStats
+}
+
+func (d *DbStats) GSIStats(queryName string) *QueryStat {
```

Comment:
Should be named QueryStats, since it's also used for view stats.

Reply:
Very good point